pyproject.toml

```diff
@@ -5,19 +5,13 @@ requires = ["maturin>=0.14,<0.15"]
 [project]
 authors = [{name = "Eventual Inc", email = "daft@eventualcomputing.com"}]
 dependencies = [
-  "numpy < 1.24",
```

Comment:
instead of deleting these they should be moved in `project.optional-dependencies`

Reply:
Added as separate extras and hooked into `all` as well